encoder/src/main/java/openlr/encoder/routesearch/SecondShortestRouteChecker.java

```diff
@@ -0,0 +1,192 @@
+package openlr.encoder.routesearch;
+
+import openlr.encoder.OpenLREncoderProcessingException;
+import openlr.map.GeoCoordinates;
+import openlr.map.Line;
+import openlr.map.utils.GeometryUtils;
+import openlr.map.utils.PQElem;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * <h1>Verifying the existence of Fully or partially joined second shortest route with length less than relative threshold</h1>
+ */
+public class SecondShortestRouteChecker {
+    private GeoCoordinates destination;
+    private List<? extends Line> location;
+    private int maxLengthAllowed;
+    private double relativeThreshold;
+
+
+    private SecondShortestRouteChecker(GeoCoordinates destination, List<? extends Line> location, int maxLengthAllowed, double relativeThreshold) {
+        this.destination = destination;
+        this.location = location;
+        this.maxLengthAllowed = maxLengthAllowed;
+        this.relativeThreshold = relativeThreshold;
+    }
+
+
+    /**
+     *
+     * @param location route
+     * @param relativeThreshold threshold relative to the route length
+     * @return Instance of SecondShortestRouteChecker for the location with threshold
+     * @throws OpenLREncoderProcessingException
+     */
+    public static SecondShortestRouteChecker on(List<? extends Line> location, double relativeThreshold) throws OpenLREncoderProcessingException {
+        if (location.isEmpty()) {
+            throw new OpenLREncoderProcessingException(OpenLREncoderProcessingException.EncoderProcessingError.NO_ROUTE_FOUND_ERROR);
+        }
+        GeoCoordinates destinationStart = location.get(location.size() - 1).getStartNode().getGeoCoordinates();
+        int locationLength = location.stream().mapToInt(Line::getLineLength).sum();
+
+        int maxLengthAllowed = locationLength + (int) (locationLength * relativeThreshold);
+
+        return new SecondShortestRouteChecker(destinationStart, location, maxLengthAllowed, relativeThreshold);
+    }
+
+    /**
+     *
+     * @param lengthAlongSecondShortestRoute length of the second shortest route
+     * @param lengthAlongLocation length of the route along location to which the second shortest route gives alternative to
+     * @return true: if the length of the second shortest route is not greater than route along location by relative threshold percentage
+     *         false: if the length of the second shortest route is greater than route along location by relative threshold percentage
+     */
+    boolean VerifyThreshold(int lengthAlongSecondShortestRoute, int lengthAlongLocation){
```

Comment:
you can add the `private` accessor here

Reply:
Also method names should begin with a lower case letter.

Reply:
00c70cf1bd4f3e32f47bccb0523ebe37cbbce6e8